lib/compat/wordpress-6.8/class-gutenberg-rest-post-types-controller-6-8.php

```diff
@@ -6,56 +6,36 @@
  */
 
 /**
- * Gutenberg_REST_Post_Types_Controller_6_8 class
+ * Register a REST field for the default rendering mode of a post type.
  *
- * Add Block Editor default rendering mode to the post type response
- * to allow enabling/disabling at the post type level.
+ * Note: Logic will become part of the `prepare_item_for_response` method when backporting to the core.
+ *
+ * @return void
  */
-class Gutenberg_REST_Post_Types_Controller_6_8 extends WP_REST_Post_Types_Controller {
-	/**
-	 * Add Block Editor default rendering mode setting to the response.
-	 *
-	 * @param  WP_Post_Type    $item    Post type object.
-	 * @param  WP_REST_Request $request Request object.
-	 * @return WP_REST_Response Response object.
-	 */
-	public function prepare_item_for_response( $item, $request ) {
-		$response = parent::prepare_item_for_response( $item, $request );
-		$context  = ! empty( $request['context'] ) ? $request['context'] : 'view';
-
-		// Property will only exist if the post type supports the block editor.
-		if ( 'edit' === $context && property_exists( $item, 'default_rendering_mode' ) ) {
-			/**
-			 * Filters the block editor rendering mode for a post type.
-			 *
-			 * @since 6.8.0
-			 * @param string       $default_rendering_mode Default rendering mode for the post type.
-			 * @param WP_Post_Type $post_type              Post type name.
-			 * @return string Default rendering mode for the post type.
-			 */
-			$rendering_mode = apply_filters( 'post_type_default_rendering_mode', $item->default_rendering_mode, $item );
+function gutenberg_editor_rendering_mode_field() {
+	register_rest_field( 'type', 'default_rendering_mode', array(
+		'get_callback' => static function( $object ) {
+			$post_type_object = get_post_type_object( $object['slug'] );
 
-			/**
-			 * Filters the block editor rendering mode for a specific post type.
-			 * Applied after the generic `post_type_default_rendering_mode` filter.
-			 *
-			 * The dynamic portion of the hook name, `$item->name`, refers to the post type slug.
-			 *
-			 * @since 6.8.0
-			 * @param string       $default_rendering_mode Default rendering mode for the post type.
-			 * @param WP_Post_Type $post_type              Post type object.
-			 * @return string Default rendering mode for the post type.
-			 */
-			$rendering_mode = apply_filters( "post_type_{$item->name}_default_rendering_mode", $rendering_mode, $item );
+			// Property will only exist if the post type supports the block editor.
+			if ( ! $post_type_object || ! isset( $post_type_object->default_rendering_mode ) ) {
```

Comment:
Still not a fan of a) artificially adding properties to the `WP_Post_Type` and b) the name

Reply:
What do you mean `artificially`? This is the only way to expose new properties in response.

Agreed on the name, I'll create a new discussion for it + if we should migrate to `supports`. This PR here is just refactoring.

Reply:
> What do you mean `artificially`? This is the only way to expose new properties in response.

When I worked on `WP_Post_Type` it wasn't really the intention to allow passing non-default fields to post type registration.

Reply:
In my mind post type supports should be for actual supports. Not for setting defaults. 

Even if a post type doesn't support this, the user can still get into this mode by simple selecting the "Show Template" option that we surface in two places in the UI. 

That is a very weakly held opinion and if you all think it should be a support I won't block that. I just want to be clear that removing the support unlike virtually every other support that I know of doesn't actually remove that feature, It just doesn't have the option in the UI checked by default. But the feature is still very much there

Reply:
It doesn't have to be a new support, it can be a configuration option of the `editor` support.

Reply:
I do like the idea of that!